pkg/reconciler/parallel/parallel.go

```diff
@@ -70,10 +70,6 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, p *v1.Parallel) pkgrecon
 	//     2.2 create a Subscription to the filter Channel, subscribe the subscriber and send reply to
 	//         either the branch Reply. If not present, send reply to the global Reply. If not present, do not send reply.
 	// 3. Rinse and repeat step #2 above for each branch in the list
-	if p.DeletionTimestamp != nil {
-		// Everything is cleaned up by the garbage collector.
-		return nil
-	}
```

Comment:
No need for this. Reconciler generated code already checks for the object status and do not call `ReconcileKind` when it is being deleted.